pkg/solana/fees/block_history.go

```diff
@@ -99,28 +99,39 @@ func (bhe *blockHistoryEstimator) readRawPrice() uint64 {
 }
 
 func (bhe *blockHistoryEstimator) calculatePrice(ctx context.Context) error {
+	switch {
+	case bhe.cfg.BlockHistorySize() == 1:
+		return bhe.calculatePriceFromLatestBlock(ctx)
+	case bhe.cfg.BlockHistorySize() > 1:
+		return bhe.calculatePriceFromMultipleBlocks(ctx, bhe.cfg.BlockHistorySize())
+	default:
+		return fmt.Errorf("invalid block history depth: %d", bhe.cfg.BlockHistorySize())
```

Comment:
I think it might be better to validate the `BlockHistorySize` during initialization so we fail fast and problems with configs can be noticed quicker. So I guess just a `if BlockHistorySize < 1 { error }`